.github/workflows/bench.yml

```diff
@@ -1,52 +1,159 @@
 name: "Run benchmark"
-
 on:
   pull_request_target:
     types: [assigned, opened, synchronize, reopened, edited]
   push:
     branches:
-      - main
-
+      - benchmarks_in_parallel
 permissions:
   contents: write
   pull-requests: write
   issues: write
-
 jobs:
-  build:
+  benchmark:
     runs-on: benchmarking-runner
     if: github.event.head_commit.message != 'Update performance results in README.md'
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    strategy:
+      matrix:
+        service: [apollo_server, caliban, netflix_dgs, gqlgen, tailcall, async_graphql, hasura, graphql_jit]
     steps:
       - name: Checkout (GitHub)
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.sha }}
-
-      - name: Build devcontainer and run benchmarks
+      
+      - name: Build devcontainer and run setup and benchmark
         uses: devcontainers/ci@v0.3
         with:
           imageName: graphql-benchmarks
           push: never
           runCmd: |
+            echo "Running setup script"
             bash ./setup.sh
-            bash ./run_benchmarks.sh
+            echo "Setup completed"
+            echo "Running benchmark for ${{ matrix.service }}"
+            chmod +x ./run_benchmarks.sh
+            valid_services=("apollo_server" "caliban" "netflix_dgs" "gqlgen" "tailcall" "async_graphql" "hasura" "graphql_jit")
+            if [[ ! " ${valid_services[@]} " =~ " ${{ matrix.service }} " ]]; then
+                echo "Error: Invalid service name '${{ matrix.service }}'"
+                exit 1
+            fi
+            if ! ./run_benchmarks.sh ${{ matrix.service }}; then
+                echo "Error: Benchmark failed for ${{ matrix.service }}"
+                exit 1
+            fi
+            echo "Benchmark completed for ${{ matrix.service }}"
+
+
+      - name: List benchmark files
+        run: |
+          echo "Contents of current directory:"
+          ls -la
+          echo "Matching bench*.txt files:"
+          ls -la bench*.txt || echo "No matching files found"
+
+          
+      - name: Upload benchmark results
+        uses: actions/upload-artifact@v3
+        with:
+          name: benchmark-results
+          path: bench*.txt
+
+
+  analyze:
+    needs: benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout (GitHub)
+        uses: actions/checkout@v4
+      
+      - name: Download all benchmark results
+        uses: actions/download-artifact@v3
+        with:
+          name: benchmark-results
+          path: .
+      
+      - name: List downloaded artifacts
+        run: ls -R
+      
+      - name: Install gnuplot
+        run: sudo apt-get update && sudo apt-get install -y gnuplot
+
+      - name: Check gnuplot installation
```

Comment:
We want to keep the CI simple and move the logic to scripts preferrably. I think this logic is already in analyse.sh.

Reply:
alright